app/models/client.rb

```diff
@@ -315,15 +358,23 @@ def unavailable(permanent:false, contact_id:nil)
     active_match = client_opportunity_matches.active.first
     Client.transaction do
       if active_match.present?
+        active_match.canceled!(contact_id)
```

Comment:
`canceled!` is called in a bunch more places, they'll all need `contact_id` passed in.

Reply:
ok.  Will search them out